wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift

```diff
@@ -569,13 +575,33 @@ final class ClientListViewController: UIViewController,
             self.clients = results
             self.selfClient = results.first(where: { $0.isSelfClient() })
             refreshViews()
+            completed?()
         }
     }
 
     @MainActor
     func refreshViews() {
         clientsTableView?.reloadData()
     }
+
+    private func updateE2EIdentityCertificateInDetailsView() {
+        guard let selectedDeviceInfoViewModel = selectedDeviceInfoViewModel else {
+            return
+        }
+        let client = if selectedDeviceInfoViewModel.isSelfClient {
+            selfClient
+        } else {
+            clients.first(
+                where: {
+                    guard let userClient = selectedDeviceInfoViewModel.userClient as? UserClient else {
+                        return false
+                    }
+                return $0.clientId == userClient.clientId
+            })
+        }
+        guard let client else { return }
+        selectedDeviceInfoViewModel.update(from: client)
+    }
```

Comment:
The func has two tasks, first find the client and then update. Therefore I split it in two funcs.
Also the code improves from my point of view:
- avoid the formatting `let client = if` which is a hard to read with this formatting.
- if `selectedDeviceInfoViewModel.userClient as? UserClient` is `nil` it would still iterate over all client, which is not necessary.

Suggestion
```swift
    private func updateE2EIdentityCertificateInDetailsView() {
        guard let client = findE2EIdentityCertificateClient() else { return }
        selectedDeviceInfoViewModel?.update(from: client)
    }

    private func findE2EIdentityCertificateClient() -> UserClient? {
        if selectedDeviceInfoViewModel?.isSelfClient == true {
            return selfClient
        }

        guard let selectedUserClient = selectedDeviceInfoViewModel?.userClient as? UserClient else {
            return nil
        }

        return clients.first { $0.clientId == selectedUserClient.clientId }
    }
```